src/Alias.ts

```diff
@@ -98,17 +98,17 @@ function applyAliases(origCommand: string, depth = 0, currentlyProcessingAliases
   // (unless there are any reference loops or the reference chain is too deep)
   const localAlias = Aliases.get(commandArray[0]);
   const localrule = commandArray[0] + "->" + localAlias + "(l)";
-  if (localAlias && !currentlyProcessingAliases.includes(localrule)) {
-    const appliedAlias = applyAliases(localAlias, depth + 1, [localrule, ...currentlyProcessingAliases]);
+  if (localAlias && !currentlyProcessingAliases.has(localrule)) {
+    const appliedAlias = applyAliases(localAlias, depth + 1, new Set([localrule, ...currentlyProcessingAliases]));
```

Comment:
I really wanted to just merge, but I can't let this go... XD

Don't create a brand-new Set, that will be the same O(N^2) behavior as the array you just replaced. Add the item to the set you have.

Reply:
I need to think this through, I'm just in love with immutables. But you're right ofc that this is a little expensive. 
For the global rules you have to copy, since a global should be able to be applied multiple times. For the local one it might suffice to just add it.

Reply:
If you really can't stand to do mutable stuff, I'm fine with you changing it back to an array. There's no point in doing something more verbose if you're going to be N^2 anyway.

Reply:
Oh don't merge, there is another bug that will make local rules be applied globally I believe. Sorry that I haven't thought this through carefully before. I was just focused on this single bug.

Reply:
It shouldn't be hard to fix though, but I'd like to write some more tests to catch it.

Reply:
based attitude

Reply:
There is a point in doing it more verbose: readability. Sets, in contrasts to Arrays, are unique and often used for lookup. So it helps with getting the codes intention.

I think mutating the set for the local aliases is totally fine, but copying is needed for the global ones. I don't believe you can circumvent that.

I will probably finish this tomorrow. It is easy to mess up.
After global rules are applied, subsequent calls to applyAlias should just consider global rules with the exception of a global rule modifying the first token/command.  I believe the function might need some restructuring of the execution flow to handle all edge cases.

Reply:
Regardless of how you end up restructuring, the following should work for mutation without copying: Add the key to the set immediately before recursion, and remove it immediately after. 

A quick sketch of why this works: You abort if the key was already present, so there's never an issue of double-removing a key. Thus, the removal always properly reverts to the prior state.

Reply:
Ah that's a great approach. I'll probably use that, but then also "unify" the application of local and global rules inside one loop / reduce, to avoid certain edge cases.